### PR TITLE
[blosc] Update to 1.21.3

### LINKED ISSUE #28877: [blosc] Update to 1.21.2
Library name: blosc

New version number: 1.21.2

Other information that may be useful (release notes, etc...)

### PR DESCRIPTION
**Pull request by wzj219**
- [X] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)
- [X] SHA512s are updated for each updated download
- [ ] ~~The "supports" clause reflects platforms that may be fixed by this new version~~
- [ ] ~~Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.~~
- [X] Any patches that are no longer applied are deleted from the port's directory.
- [X] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [X] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by JonLiu1993** [comment]
@wzj219, Thanks for your PR, please run command "./vcpkg x-add-version blosc --overwrite-version" and commit again.

**Review by JonLiu1993 — DISMISSED** [review]

File: ports/blosc/0001-find-deps.patch [hunk 1]
```diff
@@ -1,12 +1,14 @@
...
-@@ -1,10 +1,13 @@
+@@ -1,12 +1,15 @@
 -find_path(LZ4_INCLUDE_DIR lz4.h)
 +find_path(LZ4_INCLUDE_DIRS NAMES lz4.h)
  
--find_library(LZ4_LIBRARY NAMES lz4)
+ # On Windows, the lz4 library is called liblz4.lib, which is not
+ # found by using the lz4 name.
+-find_library(LZ4_LIBRARY NAMES lz4 liblz4)
 +find_library(LZ4_LIBRARY_DEBUG NAMES lz4d)
```
Comment (LilyWangLL) [inline ports/blosc/0001-find-deps.patch:12]: Here should use `find_package(lz4 CONFIG REQUIRED)` instead of `find_library` and `find_path`.

**Review by LilyWangLL — CHANGES_REQUESTED** [review]

File: ports/blosc/0002-export-blosc-config.patch [hunk 2]
```diff
@@ -51,16 +51,38 @@ index f4e5c29..24641ad 100644
...
+-        install(TARGETS blosc_static COMPONENT DEV)
++        install(TARGETS blosc EXPORT blosc-config LIBRARY COMPONENT lib)
      endif(BUILD_STATIC)
  endif(BLOSC_INSTALL)
 +
 +install(EXPORT blosc-config DESTINATION share/cmake/blosc)
+diff --git a/tests/fuzz/CMakeLists.txt b/tests/fuzz/CMakeLists.txt
+index 81000dc..cf9f909 100644
+--- a/tests/fuzz/CMakeLists.txt
++++ b/tests/fuzz/CMakeLists.txt
```
Comment (dg0yt) [inline ports/blosc/0002-export-blosc-config.patch:77]: This directory names sounds like it should be removed from the build, not patched.

File: ports/blosc/0002-export-blosc-config.patch [hunk 3]
```diff
@@ -51,16 +51,38 @@ index f4e5c29..24641ad 100644
...
--        install(TARGETS blosc_shared LIBRARY DESTINATION ${lib_dir} ARCHIVE DESTINATION ${lib_dir} RUNTIME DESTINATION bin COMPONENT LIB)
-+        install(TARGETS blosc EXPORT blosc-config RUNTIME DESTINATION bin ARCHIVE DESTINATION lib LIBRARY DESTINATION lib)
+-        install(TARGETS blosc_shared
+-                LIBRARY COMPONENT LIB
+-                ARCHIVE COMPONENT DEV
+-                RUNTIME COMPONENT LIB)
++        install(TARGETS blosc EXPORT blosc-config
++                LIBRARY COMPONENT lib
++                ARCHIVE COMPONENT lib
++                RUNTIME COMPONENT bin)
```
Comment (dg0yt) [inline ports/blosc/0002-export-blosc-config.patch:65]: I don't think we have to patch the names of the `COMPONENT`. It is not `DESTINATION`.

File: ports/blosc/0001-find-deps.patch [hunk 4]
```diff
@@ -50,3 +52,137 @@ index 688d4d5..c8b9a05 100644
...
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index 54d4817..9b7203d 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -134,7 +134,7 @@ include(GNUInstallDirs)
+ 
+ if(NOT DEACTIVATE_LZ4)
+     if(PREFER_EXTERNAL_LZ4)
+-        find_package(LZ4)
++        find_package(lz4 CONFIG REQUIRED)
```
Comment (dg0yt) [inline ports/blosc/0001-find-deps.patch:64]: `find_package(LZ4 NAMES lz4 CONFIG REQUIRED)` for getting `LZ4_FOUND`
plus setup of related variables and targets as used in blosc.

The changes to `FindLZ4.cmake` become obsolete with this change.

File: ports/blosc/0002-export-blosc-config.patch [hunk 5]
```diff
@@ -51,16 +51,38 @@ index f4e5c29..24641ad 100644
...
  
  # install
- if(BLOSC_INSTALL)
-     install(FILES blosc.h blosc-export.h DESTINATION include COMPONENT DEV)
+@@ -230,12 +230,14 @@ if(BLOSC_INSTALL)
+     install(FILES blosc.h blosc-export.h
+             DESTINATION ${CMAKE_INSTALL_INCLUDEDIR} COMPONENT DEV)
      if(BUILD_SHARED)
--        install(TARGETS blosc_shared LIBRARY DESTINATION ${lib_dir} ARCHIVE DESTINATION ${lib_dir} RUNTIME DESTINATION bin COMPONENT LIB)
-+        install(TARGETS blosc EXPORT blosc-config RUNTIME DESTINATION bin ARCHIVE DESTINATION lib LIBRARY DESTINATION lib)
```
Comment (dg0yt) [inline ports/blosc/0002-export-blosc-config.patch:58]: I can't see that upstream renamed the library target.

File: ports/blosc/0001-find-deps.patch [hunk 6]
```diff
@@ -50,3 +52,137 @@ index 688d4d5..c8b9a05 100644
...
+diff --git a/blosc/CMakeLists.txt b/blosc/CMakeLists.txt
+index 392ad1d..495f526 100644
+--- a/blosc/CMakeLists.txt
++++ b/blosc/CMakeLists.txt
+@@ -9,7 +9,7 @@ endif(HIDE_SYMBOLS)
+ 
+ # includes
+ set(BLOSC_INCLUDE_DIRS ${BLOSC_INCLUDE_DIRS} ${CMAKE_CURRENT_SOURCE_DIR})
+-if(NOT DEACTIVATE_LZ4)
++if(0)
```
Comment (dg0yt) [inline ports/blosc/0001-find-deps.patch:104]: Why do these spots need to be patched?

Comment (wzj219) [reply ports/blosc/0001-find-deps.patch:104]: The find_package already contains the process of adding header files of lz4, so there is no need to add header files separately. This code is no longer needed.

File: ports/blosc/0002-export-blosc-config.patch (hunk 5 above)
Comment (wzj219) [reply ports/blosc/0002-export-blosc-config.patch:58]: The library file name generated in the blosc package has been modified.

File: ports/blosc/0001-find-deps.patch (hunk 6 above)
Comment (dg0yt) [reply ports/blosc/0001-find-deps.patch:104]: This doesn't require a patch. The patch is needed only if we can't control `DEACTIVATE_LZ4` via CMake parameters.

File: ports/blosc/0002-export-blosc-config.patch [hunk 7]
```diff
@@ -1,8 +1,149 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index 54d4817..20f49af 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -98,9 +98,9 @@ option(BUILD_STATIC
+ option(BUILD_SHARED
+     "Build a shared library version of the blosc library." ON)
+ option(BUILD_TESTS
+-    "Build test programs from the blosc compression library" ON)
++    "Build test programs from the blosc compression library" OFF)
```
Comment (dg0yt) [inline ports/blosc/0002-export-blosc-config.patch:10]: Really NO to this patch. This is explicitly an option, i.e. you pass the desired value in `OPTIONS` of vcpkg_cmake_configure`.

File: versions/b-/blosc.json [hunk 8]
```diff
@@ -1,5 +1,15 @@
...
+    {
+      "git-tree": "93b5139c66ac3c1d789a887959762876154a1ae6",
+      "version": "1.21.3",
+      "port-version": 1
+    },
+    {
+      "git-tree": "08be7493a8b0644853e545f50d243680b1d13fd9",
+      "version": "1.21.3",
+      "port-version": 0
+    },
```
Comment (JonLiu1993) [inline versions/b-/blosc.json:12]: @wzj219, we only allow adding one "git-tree" record, please delete these two records and re-run `./vcpkg x-add-version blosc --overwrite-version` then commit again.

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

**Review by LilyWangLL — APPROVED** [review]

**Review by JonLiu1993 — APPROVED** [review]

**Comment by wzj219** [comment]
@microsoft-github-policy-service agree

**Comment by BillyONeal** [comment]
Thanks for the update!

**Pull request merged**